src/schemathesis/hooks.py

```diff
@@ -231,10 +233,10 @@ def before_add_examples(context: HookContext, examples: List[Case]) -> None:
 
 
 @HookDispatcher.register_spec([HookScope.GLOBAL])
-def add_case(context: HookContext, case: Case) -> Case:
-    """Creates an additional test per endpoint.
+def add_case(context: HookContext, case: Case, case_response: Union[Response, WSGIResponse]) -> Union[None, Case]:
```

Comment:
We have `schemathesis.utils.GenericResponse` for `Union[Response, WSGIResponse]` and `Union[None, Case]` can be replaced with `Optional[Case]` which is more concise